File: ley_ohm.js
// Importamos el módulo readline para interactuar con el usuario
const readline = require('readline');

// Creamos la interfaz de readline para interactuar con el usuario
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Solicitamos el valor del voltaje al usuario
rl.question('Ingrese el valor del voltaje del circuito: ', (Voltaje) => {
    // Solicitamos el valor de la resistencia
    rl.question('Ingrese el valor de la resistencia del circuito: ', (Resistencia) => {
        // Convertimos los valores a enteros y calculamos la intensidad (amperaje)
        const intensidad = parseInt(Voltaje) / parseInt(Resistencia);

        // Imprimimos el resultado en consola
        console.log(`Al conectar un resistor de R ${Resistencia} ohm a una fuente de V ${Voltaje} voltios circulará una corriente de ${intensidad} amperios.`);

        // Cerramos la interfaz readline
        rl.close();
    });
});

File: ley_ohm.cpp
#include <iostream>
using namespace std;

void voltaje(){
    double corriente, resistencia;
    cout <<"ingrese la corriente (I) en ampérios: ";
    cin >>corriente;
    cout <<"ingrese la resisteencia (R) en ohmios: ";
    cin>> resistencia;
    cout<<"el voltaje V es: "<<corriente*resistencia<<"v"<<endl;
}
void corriente(){
    double voltaje, resistencia;
    cout<<"ingrese un voltaje en voltios: ";
    cin>>voltaje;
    cout<<"ingrese la resistenci en ohmios: ";
    cin>>resistencia;
    cout<<"la corriente I es"<<voltaje/resistencia<<"ohmios"<<endl;
}
void resistencia(){
    double voltaje, corriente;
    cout<<"Ingrese el voltaje en voltios: ";
    cin>>voltaje;
    cout<<"ingrese la corriente I en amperios: ";
    cin>>corriente;
    cout<<"la resistencia R es: "<<voltaje/corriente<< "ohmios"<<endl;
}
int main(){
    int opcion;
    cout<<"===calculadora de la ley de ohm===\n";
    cout<<"1. Voltaje\n";
    cout<<"2. Corriente\n";
    cout<<"3. Resistencia\n";
    cout<<"Seleciione una opcion de 1 a 3: ";
    cin>>opcion;

    switch (opcion)
    {
    case 1:voltaje();break;
    case 2:corriente();break;
    case 3:resistencia();break;
    default:cout<<"opcion no valida,"<<endl;
    }
    return 0;
}